x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_logic.ts

```diff
@@ -187,4 +220,123 @@ export const ResultSettingsLogic = kea<MakeLogicType<ResultSettingsValues, Resul
       },
     ],
   }),
+  selectors: ({ selectors }) => ({
+    resultFieldsAtDefaultSettings: [
+      () => [selectors.resultFields],
+      (resultFields: FieldResultSettingObject) => areFieldsAtDefaultSettings(resultFields),
+    ],
+    resultFieldsEmpty: [
+      () => [selectors.resultFields],
+      (resultFields: FieldResultSettingObject) => areFieldsEmpty(resultFields),
+    ],
+    stagedUpdates: [
+      () => [selectors.lastSavedResultFields, selectors.resultFields],
+      (lastSavedResultFields: FieldResultSettingObject, resultFields: FieldResultSettingObject) =>
+        !isEqual(lastSavedResultFields, resultFields),
+    ],
+    reducedServerResultFields: [
+      () => [selectors.serverResultFields],
+      (serverResultFields: ServerFieldResultSettingObject) =>
+        Object.entries(serverResultFields).reduce(
+          (acc: ServerFieldResultSettingObject, [fieldName, resultSetting]) => {
+            if (resultSetting.raw || resultSetting.snippet) {
+              acc[fieldName] = resultSetting;
+            }
+            return acc;
+          },
+          {}
+        ),
```

Comment:
Functional programming is funny! Here's an alternative approach:

```
Object.entries(serverResultFields)
  .filter(([_, resultSetting]) => resultSetting.raw || resultSetting.snippet)
  .reduce((acc, [fieldName, resultSetting]) => {...acc, [fieldName]: resultSetting}, {})
```

I don't think you need to change it,  they both accomplish the same thing and are about as readable, just wanted to flex a little 💪